packages/pds/src/app-migrations/remove-declarations.ts

```diff
@@ -0,0 +1,78 @@
+import { chunkArray } from '@atproto/common'
+import AppContext from '../context'
+import Database from '../db'
+import { appMigration } from '../db/leader'
+import { MessageDispatcher } from '../event-stream/message-queue'
+import { PreparedDelete, prepareDelete } from '../repo'
+import { RepoService } from '../services/repo'
+
+const MIGRATION_NAME = '2023-03-14-remove-declarations'
+const SHORT_NAME = 'remove-declarations'
+
+export async function removeDeclarationsMigration(ctx: AppContext) {
+  await appMigration(ctx.db, MIGRATION_NAME, (tx) => main(tx, ctx))
+}
+
+async function main(tx: Database, ctx: AppContext) {
+  console.log(SHORT_NAME, 'beginning')
+  tx.assertTransaction()
+  const now = new Date().toISOString()
+
+  // The message dispatcher usually ensures that the app view indexes these updates,
+  // but that has been taken care of via a db migration to remove the indexes entirely.
+  const noopDispatcher = new MessageDispatcher()
+  noopDispatcher.destroy()
+
+  const repoTx = new RepoService(
+    tx,
+    ctx.repoSigningKey,
+    noopDispatcher,
+    ctx.blobstore,
+  )
+
+  // For each user remove declaration, assertion, confirmation records.
+
+  // Should be 4-5k records
+  const recordsToDelete = await tx.db
+    .selectFrom('record')
+    .innerJoin('repo_root', 'repo_root.did', 'record.did') // Ignore any records not in a repo
```

Comment:
did you encounter any of these in your tests?

Reply:
There were 7, I think.  I looked at them all and they were indeed malformed in some way (either a handle rather than a did, or empty did).

Reply:
I didn't check actually, but I could do a run without this and find out.

Reply:
Okay this was overly paranoid, it just doesn't happen on any environment 😆 Will remove.